### PR TITLE
use XHRs to fetch dashboard data

### PR DESCRIPTION
**Pull request by mozkeeler**
It maybe needs a bit of polish (e.g. the "minimum issuance" slider has no effect on the "worst CAs" chart - that always stays as "worst CAs with issuance > 1000"), but the functionality is basically there.

### CODE REVIEW

**Comment by monicachew** [comment]
Cool! This is a good approach. Is this live at people.mozilla.org/~dkeeler/dashboard?

**Comment by mozkeeler** [comment]
Yes - what's there should be up-to-date.

File: dashboard/db2json.js [hunk 1]
```diff
@@ -91,9 +87,52 @@ function completionDump(name, issuerArray) {
...
 }
 
-// An array of issuers
+function initIssuerData(path) {
+  if (fs.existsSync(path)) {
+    fs.unlinkSync(path);
+  }
+}
+
+function dumpScoresAndVolumeForIssuer(issuer, issuerFilename) {
```
Comment (monicachew) [inline dashboard/db2json.js]: We could use some documentation of the JSON blob here

File: dashboard/index.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+function escapeName(name) {
+  return name.replace(/[^A-Za-z0-9]/g, "_");
+}
+
+function getChartData(name, continuation) {
+  let escapedName = escapeName(name);
+  let req = new XMLHttpRequest();
+  req.open("GET", "data/" + escapedName + ".json", true);
+  req.onreadystatechange = function() {
+    if (req.readyState == 4 && req.status == 200) {
```
Comment (monicachew) [inline dashboard/index.js]: Does XMLHttpRequest.DONE work instead of "4"?

File: dashboard/index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    getChartData(name, function(series) {
+      new Highcharts.StockChart({
+        chart: {
+          renderTo: "timeseries"
+        },
+        legend: commonLegend,
+        series: series,
+        yAxis: [{
+          max: 1.0,
+          min: 0.0
```
Comment (monicachew) [inline dashboard/index.js]: Looks like you can set global chart options to reduce some verbiage here.

http://www.highcharts.com/docs/getting-started/how-to-set-options#2

(we also use this in the pinning dashboard in chartoptions.js)

**Comment by monicachew** [comment]
Looks great! Just a few minor nits.

**Comment by mozkeeler** [comment]
Thanks for the review! I addressed your comments and re-deployed to people.mozilla.org/~dkeeler/dashboard

File: dashboard/index.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        legend: commonLegend,
+        series: series,
+        yAxis: commonYAxis
+      });
+    });
+  }
+}
+
+// Strangely, it looks like passing in values for legend and yAxis don't work,
+// so they have to be specified with each chart created.
```
Comment (monicachew) [inline dashboard/index.js]: oh weird :(

**Pull request merged**